src/package/model.rs

```diff
@@ -0,0 +1,316 @@
+use std::time::SystemTime;
+
+use actix::prelude::*;
+use diesel::{
+    self,
+    pg::upsert::{excluded, on_constraint},
+    prelude::*,
+};
+use failure::Error;
+
+use crate::schema::{dependencies, package_groups, packages, versions};
+use crate::user::model::LookupUser;
+use crate::util::Database;
+
+#[derive(Clone)]
+pub struct PackageName {
```

Comment:
A few comments here:

- PackageName just looks like Name from elba
- PackageVersion feels a bit redundant - is this not replaceable with elba's Summary or something like that?
- Can we just use `semver::Version` instead of `String` for the semver fields?